engine/pkg/rpcerror/normalize.go

```diff
@@ -0,0 +1,212 @@
+// Copyright 2022 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package rpcerror
+
+import (
+	"encoding/json"
+	"fmt"
+	"reflect"
+	"strings"
+	"sync"
+
+	"github.com/pingcap/errors"
+	"go.uber.org/zap"
+	"google.golang.org/grpc/codes"
+
+	"github.com/pingcap/tiflow/dm/pkg/log"
+	pb "github.com/pingcap/tiflow/engine/enginepb"
+)
+
+type normalizeOpts struct {
```

Comment:
I don't think we need to define a new type `normalizeOpts` because the fields of it are the same as `Prototype[E errorInfo]`.
Why not reuse them directly?

Reply:
`Prototype` has type paramaters, which can make defining `NormalizeOpt` tricky.